### ISSUE TITLE
Calculate from choice filtered or_other value

### ISSUE THREAD

**Issue by DShokes** [Esri/Survey123Community#338]
Trying to make solution mixing calculator and running into many problems. The workflow is the user enters a total volume. Water and a primary product are populated automatically. The user can add any number of additional products that each have a set rate per gallon. Since repeats have known bugs, I am allowing up to three additional products.

If the user selects a defined product and clicks the filtered rate. The calculation works fine.
![image](https://cloud.githubusercontent.com/assets/3903212/13303923/94d4e022-db17-11e5-966c-d17d4791c9d1.png)

However, the calculation doesn't work if other is chosen. 
![image](https://cloud.githubusercontent.com/assets/3903212/13303939/aea21a42-db17-11e5-8b99-dd70d090219e.png)
This is what I'm using for the formula: round ( ( number(${product_rate3}) * number(${tot_volume}) ), 2)
Another issue is that the filter choices and or_other don't play along. Product Units and Product Rate can't be autopopulated if 3rd Product allows for Other values.
Here is my table:
[Mixing Form.xlsx](https://github.com/Esri/Survey123Community/files/145627/Mixing.Form.xlsx)

![image](https://cloud.githubusercontent.com/assets/3903212/13303735/a186ec76-db16-11e5-91a3-429c8a8b6f2e.png)

.
Any suggestions?

**Comment by HannahFerrier** [comment]
Please keep in mind that the values entered for **or_other** are stored in a separate data column.

From the XLS Form Help:

When you export data using this **or_other** option, in the question column, you will see the value _other_. A separate column will have the answer for the questions in which the user selected other. This makes data analysis more cumbersome, so we do not recommend the **or_other** construct for large scale data collection efforts.

**Comment by DShokes** [comment]
Thanks Hannah. So there is no way to provide a combo box (or similar functionality) that contains a list of common values, but allows for free entry?  The existence of the separate "other" field shouldn't break the logic of fields with defined references. Even if I added 'other' to the product list, the complications still trickle down while trying to automatically populate units and rates with known values. 

At the end of the day, this example highlights the huge platform-wide geodatabase limitation while entering attributes. Fields can be entered two ways: 
1. fill in the blank (possibly with a constraint such and numerical range or no nulls)
2. picking from a domain that must be updated/overwritten/republished/micromanaged for new possibilities.

Survey123 seems like the great opportunity to overcome this limitation.  The solution that comes to mind is a combobox which may or may not be dynamically appended with new unique values.

**Comment by DShokes** [comment]
I found a workaround around. Turns out that auto-created "other" fields are named by adding "_other" to the derived field name; in my case, "additional_products_other". I made a template that shows automated calculations on fields if their group is relevant. Feel free to use it as an example of using relevant and calculations. I only wish that it worked as a repeat instead of 3 max.

[Example Mix Form.xlsx](https://github.com/Esri/Survey123Community/files/151370/Example.Mix.Form.xlsx)

[Example Mix Form.zip (My Survey Designs Folder)](https://github.com/Esri/Survey123Community/files/151368/Example.Mix.Form.zip)
